compiler/rustc_codegen_llvm/src/back/lto.rs

```diff
@@ -92,6 +81,13 @@ fn prepare_lto(
                                             static library outputs",
                 );
                 return Err(e);
+            } else if *crate_type == CrateType::Dylib {
+                if !cgcx.unstable_options {
+                    return Err(diag_handler.fatal(
+                        "LTO cannot be used for `dylib` crate type
+ without `-Zunstable-options`",
```

Comment:
I don't think this is enough of a gate. Can we add a dedicated -Z flag for this support, so folks don't accidentally enable it (due to wanting something else unstable)?

Reply:
Makes sense